### PR TITLE
feat: applets to quickly launch applications

### PR DESCRIPTION
**Pull request by ohaiibuzzle**
tldr the idea is that the user can launch apps they want inside the bottle without going through Whisky's UI
The implementation here uses an applet with AppleScript to launch the application directly using Whisky's Wine instead of adding something like an URL handler to Whisky (which *may* be more messy)

### CODE REVIEW

**Comment by JoshuaBrest** [comment]
> tldr the idea is that the user can launch apps they want inside the bottle without going through Whisky's UI
> The implementation here uses an applet with AppleScript to launch the application directly using Whisky's Wine instead of adding something like an URL handler to Whisky (which _may_ be more messy)

We talked about this in discord but in my opinion, this would not be great.
 - When updating implementations with wine installation, the shortcut would not be updated.
   - This can include changing the wine paths and stuff.
  - User settings won't be saved (hud, esync) 
  - I think you mentioned this, but, because the implementation on how wine is launched differs, it can make make some apps not work
    - launch with diffrent libs and stuff

**Comment by ohaiibuzzle** [comment]
> When updating implementations with wine installation, the shortcut would not be updated.  

It takes Wine from the static Whisky variables, so it will always be correct

> User settings won't be saved (hud, esync)

Those are just environment vars. They can be added.

> I think you mentioned this, but, because the implementation on how wine is launched differs, it can make make some apps not work 

Uh what? Keep in mind the way the AppleScript is internally built is exactly how Apple's shell script did it, so it's a non-issue. The libs loaded are determined by dyld, not the applets

**Comment by JoshuaBrest** [comment]
> > When updating implementations with wine installation, the shortcut would not be updated.
> 
> It takes Wine from the static Whisky variables, so it will always be correct on creation

Meaning that if we change the file structure, the files won't work

> > User settings won't be saved (hud, esync)
> 
> Those are just environment vars. They can be added.

true, but i don't think you can change it because you compile the applescript.

> > I think you mentioned this, but, because the implementation on how wine is launched differs, it can make make some apps not work
> 
> Uh what? Keep in mind the way the AppleScript is internally built is exactly how Apple's shell script did it, so it's a non-issue. The libs loaded are determined by dyld, not the applets

If we change the way we launch wine, like your video, different apps could not work

**Review by JoshuaBrest — CHANGES_REQUESTED** [review]
I would like further discussion before this gets merged.

File: Whisky/Utils/IcnsWriter.swift [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        } catch {
+            if let whiskyIcon = Bundle.main.path(forResource: "AppIcon", ofType: ".icns") {
+                return URL(fileURLWithPath: whiskyIcon)
+            }
+        }
+        return nil
+    }
+
+    func convertIconSize(_ size: Int) -> UInt32 {
+        // Converts a size in pixels in a range to an appropriate icon size
```
Comment (JoshuaBrest) [inline Whisky/Utils/IcnsWriter.swift:60]: Before you merge this, I have a library that can do this.

**Pull request closed without merging**